extensions/example-extension/renderer.tsx

```diff
@@ -1,25 +1,45 @@
-import { LensRendererExtension } from "@k8slens/extensions";
-import { ExampleIcon, ExamplePage } from "./page";
+import { Component, Interface, K8sApi, LensRendererExtension } from "@k8slens/extensions";
+import { ExamplePage, ExamplePageParams, namespaceStore } from "./page";
 import React from "react";
+import path from "path";
 
 export default class ExampleExtension extends LensRendererExtension {
-  clusterPages = [
+  clusterPages: Interface.PageRegistration[] = [
     {
-      id: "example",
-      title: "Example Extension",
       components: {
-        Page: () => <ExamplePage extension={this}/>,
+        Page: (props: Interface.PageComponentProps<ExamplePageParams>) => {
+          return <ExamplePage {...props} extension={this}/>;
+        },
+      },
+      params: {
+        // setup basic param "exampleId" with default value "demo"
+        exampleId: "demo",
+
+        // setup advanced multi-values param "selectedNamespaces" with custom parsing/stringification
+        selectedNamespaces: {
+          defaultValueStringified: ["default", "kube-system"],
+          multiValues: true,
+          parse(values: string[]) { // from URL
+            return values.map(name => namespaceStore.getByName(name)).filter(Boolean);
+          },
+          stringify(values: K8sApi.Namespace[]) { // to URL
+            return values.map(namespace => namespace.getName());
+          },
+        }
       }
```

Comment:
`params` is an optional field, which is good when the page doesn't support params,  but when I tried not specifying it, I got:
```
TypeError
Cannot destructure property 'exampleId' of 'this.props.params' as it is undefined.
```

can there be some default behaviour when params is not specified such that `this.props.params` is not undefined when `ExamplePage` methods try to use it?

Reply:
Well, I guess this is intentional behaviour. If your component didn't specified any `params` why it's trying to get access to this object? Otherwise could be provided with empty `{}` in case of non-specified params, but I'd rather keep current behaviour since it pointing you that's something wrong in the component.

Reply:
Heh, yeah, I am experimenting. It's an extension dev error but it can reasonably happen because there is still some ambiguity as to what should get defined where. Wondering if we can catch such a situation in the implementation. Also trying to see if the example can be coded up to make things clearer/more robust...

Reply:
>  because there is still some ambiguity as to what should get defined where

for me everything pretty clear:
1. define `params` at page registration if needes
2. gets injected `props.params` to registered `Page` component
3. PROFIT!

All what needs to know is current interface for `props.params`. Basically it's the same defined `params` object with same keys, but instead of config values you get instances of `PageParam` with that configs. It's getter/setter helper for everything what needs to CRUD that param.

Reply:
Yes, I think you're right, the ambiguity is mostly in my head :) Because `PageParam` isn't imported directly it feels like it comes out of nowhere but that's the nature of interfaces. We'll have to carefully document `PageComponentProps` and `PageParams` for someone to understand how to use them to the full extent. 
Just need to try a few more examples...